### ISSUE TITLE
Still able to book unavailable slots

### ISSUE THREAD

**Issue by smileBeda** [SergeyMosin/Appointments#465]
I have setup 2 calendars (a target and a  source to check against).
I have selected the weekly template and setup slots.

I added a meeting in my source calendar during one of the slots - this meeting has NOT been added by the appointments app of course. It is an event I will attend personally with my family, so I add it _manually_ to my source calendar.

I can still book an appointment using the Appointments app even if the time is already used up by my event in the source calendar.

If instead I book something through the app, then that slot is unavailable after.

So it becomes impossible to make sure no one books when I am not available. I would have to edit the Target Calendar Template every time?
This makes is cumbersome.

Am I missing something? It says "check these calendars for conflicts" and I selected my source calendar there. Yet it appears to only check for events created by the app, not any other event.

**Comment by SergeyMosin** [comment]
Can you send me (sergey [at] srgdev.com) your settings dump `Appointments Settings > Advanced Settings > Debugging > Settings Dump` and the `.ics` file for the event that is supposed block the time slot, so I can try to replicate this on my testing machine.

**Comment by smileBeda** [comment]
@SergeyMosin - thanks, email sent. In further testing, it appears it is a specific type of event: added by calendly to my Mac calendar, which then was imported to the nextcloud calendar:
- use Mac Calendar for years
- have calendly add events to it through the online calendly app
- finally export Mac calendar with all events in it
- import these to nextcloud

So technically the event _is_ a Mac Calendar event but it was originally added to it by calendly. Maybe that is an issue, since it appears to work on other eventsI had added manually to my Mac calendar.

I do, however, have one more issue, which results in countless JS errors on the front end:
Uncaught ReferenceError: OC is not defined
    at index.js:25:15
Uncaught ReferenceError: OC is not defined
    at en_GB.js?v=188445fb-17:1:1
Uncaught ReferenceError: jQuery is not defined
    at script.js?v=188445fb-17:492:12
etc etc

This happens when I simply load the Appointments URL in a logged out front end.
Additionally, when booking, it does send me the email, it adds the event, but it never stops spinning in the front end, effectively leaving the user there believing it "did not work".

It appears to me the whole of jQuery is missing, somehow, but I wouldn't know why or how to fix this (well, I would know in a "normal" website, but I assume if jQuery is required it would be loaded in a no-conflict mode already by the plugin?)

Please let me know if you need me to open a new issue for this.

**Comment by SergeyMosin** [comment]
Thanks sending the `Settings Dump` and the `.ics` file. It looks like your event class is set to private ( `CLASS:PRIVATE` ) and `PRIVATE` events are ignored for security purposes, relevant code here: 
https://github.com/SergeyMosin/Appointments/blob/e179a0e08bd3f00d5be33c3a68c482ff634f7533/lib/Backend/BCSabreImpl.php#L535-L543

Different calendar apps call the `CLASS` property by different names, for example in NC it is called `When shared ...` 

![image](https://github.com/SergeyMosin/Appointments/assets/10747559/133e5a13-cdaf-43e0-b0ca-56933b57073d)

----

> Uncaught ReferenceError: OC is not defined

Some of the "heavy" NC JS files are taken out of the appointments Booking Page for faster page loads. The appointment form does not depend on the `OC` object or `jQ`

> when booking, it does send me the email, it adds the event, but it never stops spinning in the front end,

It looks like the request does get sent to the back-end but then it get stuck there (maybe waiting for the mail server connection). Is there anything in NC logs ? If you have "Email Me when appointment is booked" option enabled the email should be sent or an error with more info is logged. Is an email sent to attendee ? In any case, I'll try replicate this.

**Comment by smileBeda** [comment]
I am sorry the delay here.

While I get the point of the (external) meeting "not working" for the reason you mention, I still cannot get the online booking form to work.
1. There are are about 18 jQuery errors showing up when clicking the button to book
2. The most breaking is clearly this error: `form:1 Refused to send form data to 'https://site.tld/index.php/apps/appointments/pub/EvxC6uyBMFZg6dSaMA%3D%3D/form' because it violates the following Content Security Policy directive: "form-action 'self' https://site.tld".`
3. The event is created, the emails are sent, but the form never stops spinning, gives no "success" feedback, or, if set, redirect to any target also fails to trigger.

Basically the prospect will think it "did not work"

I do not have any related errors that appear either right while or after booking an appointment.

The refusal to send data to me seems pretty much blocking - if it refuses, it also cannot receive any success, I suppose.

**Comment by SergeyMosin** [comment]
Can you email me the URL of the page so I can try to book one test appointment

**Comment by smileBeda** [comment]
Sent the email with the publicly accessible booking form. Thank you!

**Comment by smileBeda** [comment]
Heya! Are there any fixes (available) or updates (planned) for this issue?
I would really like to move away from calendly sooner or later.

**Comment by SergeyMosin** [comment]
Bug confirmed.
> Combination of "Skip Validations Step" and "Redirect Confirmed URL" settings related to CSP. Chrome (Firefox is not effected for some reason) does not want to do a redirect to a different (sub)domain in the same call as from data submission

**Comment by SergeyMosin** [comment]
Fixed in v2.1.8
